### PR TITLE
lsp-html - update to new version of custom data and add new configuration options

### LINKED ISSUE #4327: lsp-html: update settings
Microsoft's HTML language server configuration settings have evolved (see [here](https://github.com/sublimelsp/LSP-html/blob/master/LSP-html.sublime-settings) or [here](https://github.com/microsoft/vscode/blob/main/extensions/html-language-features/package.json)).

Most notably, `html.experimental.customData` has been renamed to `html.customData`, which means `lsp-html` currently can't be used in conjunction with web components.

### PR DESCRIPTION
**Pull request by themkat**
# Updating custom data
I could not get custom data to work on my machine with a simple option/property/setting. From reading the vscode html tools source code, it seems that this extension uses the property internally in the same way we do here (read diffs in this PR). In summary:
- Send the data paths to the language server in the initialization options field.
- Handle a request for the content of these files in the client code.

When the content of these files have been sent to the server, it seems to work perfectly. I can complete, view hover information and more. I used various projects, like [the example project from the vscode team](https://github.com/microsoft/vscode-custom-data/tree/main/samples/helloworld). I added an extra custom data definition with some bogus to make it more interesting:
```json
{
  "version": 1.1,
  "tags": [
    {
      "name": "dafuq",
      "description": "The super cool new tag for the cool kids",
      "attributes": [
        { "name": "isThisShit" },
        {
          "name": "test",
          "values": [
            {
              "name": "is-working"
            }
          ]
        }
      ]
    }
  ]
}

```

image
image
image

Completion of attributes also works perfectly. I just have to be sure to activate yas-minor-mode (or similar templating system). 

Fixes #4327
(probably)

# New configuration options
Added `lsp-html-hover-documentation` and `lsp-html-hover-references`, which are both on by default (even if not set from the language server side). I think it makes sense to be able to set these, especially on older machines where I try to get more performance out of turning off features.

### CODE REVIEW

File: clients/lsp-html.el [hunk 1]
```diff
@@ -182,9 +206,13 @@ styles."
                                      (cons (lsp-package-path 'html-language-server)
                                            lsp-html-server-command-args)))
                   :activation-fn (lsp-activate-on "html")
+                  :major-modes '(html-mode)
```
Comment (jcs090218) [inline clients/lsp-html.el:209]: Why do we need this since we already have `:activation-fn`. 🤔

Comment (themkat) [reply clients/lsp-html.el]: Had some issues with lsp-mode not starting for html files and giving me an error message on no servers available. Removed the line and will investigate the issues further. Might be some weirdness with my local setup.


Removed the major-mode line and rebased 🙂

**Comment by jcs090218** [comment]
One small warning:

```
lsp-html.el:191:2: Warning: docstring wider than 80 characters
```

File: clients/lsp-html.el [hunk 2]
```diff
@@ -23,19 +23,21 @@
 ;;; Code:
 
 (require 'lsp-mode)
+(require 'f)
```
Comment (jcs090218) [inline clients/lsp-html.el:26]: I don't think require is necessary since `lsp-mode` already has it.

Comment (themkat) [reply clients/lsp-html.el]: Removed it in latest commit

**Comment by themkat** [comment]
> One small warning:
> 
> ```
> lsp-html.el:191:2: Warning: docstring wider than 80 characters
> ```

Fixed it in latest commit 🙂

**Review by jcs090218 — APPROVED** [review]

**Comment by jcs090218** [comment]
Thank you!

**Pull request merged**